6_postUpdateDom.tsx

```diff
@@ -68,22 +70,60 @@ function WidthGetter3() {
 }
 
 /*
-Protoze z DOMu jen cteme, ve skutecnosti by byl lepsi useEffect. useLayoutEffect se od nej lisi v jedinem aspektu - pousti se ve chvili, 
-kdy browser jeste nestihne prekreslit takze nase updaty se zohledni ihned. Takze abych ho trochu ospravedlnil, pridal jsem update window title :)
- */
+Because we showing width usage of useLayoutEffect is correct, because if we would like blink of initial zero value. 
+So useEffect we will show on changing browser title which is not part of DOM that Bobril controls.
+*/
 
 function WidthGetter3Bonus() {
   const [width, setWidth] = b.useState(0);
   const ref = b.useRef<b.IBobrilCacheNode>();
-  const origPageTitle = b.useState(""); //
   b.useLayoutEffect(() => {
     const elem = b.getDomNode(ref.current) as HTMLElement;
     const width = elem.getBoundingClientRect().width;
     setWidth(width);
-    if (origPageTitle() === "") origPageTitle(document.title); //
-    document.title = `width: ${width}`; //
   });
-  b.useLayoutEffect(() => () => (document.title = origPageTitle()), []); // restore original page title
+  // clear separation of concerns to individual hooks
+  b.useEffect(() => {
+    const backupTitle = document.title;
+    document.title = `width: ${width}`;
+    return () => {
+      // restore original page title
+      document.title = backupTitle;
+    };
+  }, [width]); // we need to change title only when width changes
```

Comment:
It is ok that for every `width` update we change the title twice (once back to original title, resave it to the `backupTitle` and set it to correct value)?

This should do ..
```ts
b.useEffect(() => {
  const origPageTitle = document.title;
  () => (document.title = origPageTitle);
}, []);
b.useEffect(() => {
  document.title = `width: ${width}`;
}, [width]); // we need to change title only when width changes
```

Reply:
useEffect will just schedule run. It will schedule it twice, but actual execution will be just one. You can check it with console.log

Reply:
Double scheduling is just implementation detail which could be fixed in future. Specification is that useEffect method will be run only once for frame.

Reply:
Sorry now I understand - yes it will actually return original value for each change, but that's not problem (browser will not change it synchronously anyway). But code is simpler and easily bug free. So it is not good pattern to split it.